org.opent2t.sample.binaryswitch.superpopular/com.wemo.binaryswitch/js/thingTranslator.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+var q = require('q');
+var wemoClient = require('wemo-client');
+var wemo = new wemoClient();
+var url = require('url');
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+/**
+ * Helper method to convert the provider schema to the platform schema.
+ */ 
+function providerSchemaToPlatformSchema(providerState, deviceInfo, expand) {
```

Comment:
Nit: Helpers i thought we were following the convention of pulling these at the bottom of the file + naming them starting wiht an '_'